vector/src/test/java/im/vector/app/test/fakes/FakeRegistrationWizard.kt

```diff
@@ -25,6 +25,14 @@ import org.matrix.android.sdk.api.session.Session
 class FakeRegistrationWizard : RegistrationWizard by mockk() {
 
     fun givenSuccessfulDummy(session: Session) {
-        coEvery { dummy() } returns RegistrationResult.Success(session)
+        givenSuccessFor(session) { dummy() }
+    }
+
+    fun givenSuccessFor(result: Session, expect: suspend RegistrationWizard.() -> RegistrationResult) {
+        coEvery { expect(this@FakeRegistrationWizard) } returns RegistrationResult.Success(result)
+    }
+
+    fun givenSuccessfulAcceptTerms(session: Session) {
+        coEvery { acceptTerms() } returns RegistrationResult.Success(session)
```

Comment:
Maybe we could replace it by `givenSuccessFor(session) { acceptTerms() }`? Plus, do we still need to keep `givenSuccessfulDummy` and `givenSuccessfulAcceptTerms` if we have the more generic method `givenSuccessFor`?

Reply:
aha :facepalm: you're right! and it turns out both `givenSuccessfulAcceptTerms` `givenSuccessfulDummy` aren't used because they've been replaced by the generic version, will delete the unused functions